cdap-app-fabric/src/main/java/co/cask/cdap/metadata/EntityValidator.java

```diff
@@ -65,7 +65,9 @@ public void ensureEntityExists(Id.NamespacedId entityId) throws NotFoundExceptio
     try {
       namespaceClient.get(entityId.getNamespace());
     } catch (NamespaceNotFoundException e) {
-      throw e;
+      if (!Id.Namespace.SYSTEM.equals(entityId.getNamespace())) {
```

Comment:
why not check for System namespace and altogether skip the nsClient#get call for efficiency.